packages/main/src/components/OverflowToolbar/index.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { Children, Component, CSSProperties, ReactNode, ReactNodeArray, RefObject } from 'react';
+import styles from './OverflowToolbar.jss';
+import { StyleClassHelper, withStyles } from '@ui5-webcomponents-react/base';
+import { ToolbarAlignment } from '../../lib/ToolbarAlignment';
+import { ToolbarDesign } from '../../lib/ToolbarDesign';
+import { CommonProps } from '../../interfaces/CommonProps';
+import { ClassProps } from '../../interfaces/ClassProps';
+import { Popover } from '../../lib/Popover';
+import { Button } from '../../lib/Button';
+import { PlacementType } from '../..';
+import boot from '@ui5/webcomponents-base/src/boot';
+
+export interface ToolbarPropTypes extends CommonProps {
+  width?: CSSProperties['width'];
+
+  children: ReactNode | ReactNodeArray;
+
+  align?: ToolbarAlignment;
+
+  toolbarDesign?: ToolbarDesign;
+
+  overflow?: boolean;
+}
+
+interface ToolbarInternalProps extends ToolbarPropTypes, ClassProps {}
+
+@withStyles(styles)
+export class OverflowToolbar extends Component<ToolbarInternalProps> {
+  static defaultProps = {
+    width: 'auto',
+    align: ToolbarAlignment.Start,
+    toolbarDesign: ToolbarDesign.ContentBar,
+    overflow: false
+  };
+  state = {
+    children: this.props.children,
+    toolbarWidth: 0,
+    popoverElements: [],
+    popoverOpen: true,
+    renderToggle: false,
+    previousWidth: []
+  };
+  private toolbarRef: RefObject<HTMLDivElement> = React.createRef();
+
+  componentDidMount(): void {
+    boot().then(() => {
+      if (this.props.overflow) {
+        if (this.state.toolbarWidth === 0) {
+          this.setState({ toolbarWidth: this.toolbarRef.current.scrollWidth });
+        }
+        window.addEventListener('resize', () => {
+          setTimeout(() => {
+            this.handleResize();
+          }, 50);
+        });
+      }
+    });
+  }
+
+  componentDidUpdate(prevProps: Readonly<ToolbarInternalProps>, prevState, snapshot?: any): void {
+    boot().then(() => {
+      if (this.props.overflow) {
+        if (this.toolbarRef.current.clientWidth < this.toolbarRef.current.scrollWidth) {
+          this.handleResize();
+        }
+        if (this.props.width !== prevProps.width) {
+          this.handleResize();
+        }
+      }
+    });
+  }
+
+  componentWillUnmount(): void {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  private handleResize() {
+    let newChildren = this.state.children;
+    let { popoverElements, renderToggle, previousWidth } = this.state;
```

Comment:
Please destruct the `toolbarWidth` from the state as well. And I would recommend to introduce a new variable with the value of `this.toolbarRef.current` - this will make the code much more readable